instrumentation/pekko/pekko-http-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pekkohttp/v1_0/PekkoHttpTestSetup.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pekkohttp.v1_0;
+
+import static io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerUsingTest.TEST_CLIENT_IP;
+import static io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerUsingTest.TEST_USER_AGENT;
+
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.testing.internal.armeria.client.ClientFactory;
+import io.opentelemetry.testing.internal.armeria.client.WebClient;
+import io.opentelemetry.testing.internal.armeria.client.logging.LoggingClient;
+import io.opentelemetry.testing.internal.armeria.common.HttpHeaderNames;
+import java.time.Duration;
+
+public final class PekkoHttpTestSetup {
```

Comment:
did you consider extending `AbstractHttpServerUsingTest` instead of introducing this class?

Reply:
AbstractHttpServerUsingTest carries a load of existing tests with it. I just want to add one new test that is independent of AbstractHttpServerUsingTest's existing tests.

Reply:
The WebClient setup in this class is copied from private code in this lib. If I am allowed to modify that code so that I can access then I can refactor or remove this new class.

Reply:
I think you are mixing `AbstractHttpServerUsingTest ` with `AbstractHttpServerTest`. `AbstractHttpServerUsingTest ` contains zero tests.

Reply:
The reason I need this new class is because I need a WebClient instance. AbstractHttpServerUsingTest does not create a WebClient.

HttpServerInstrumentationExtension creates a WebClient but it needs an InstrumentationTestRunner.

These are all major rabbit holes.

I can get rid of this class and just create my own  WebClient in my test class. I just didn't want to write this in Scala but I will if this Java class is a problem for you.

Reply:
Typically you'd extend the `AbstractHttpServerUsingTest` and in the test register the extension with
```java
@RegisterExtension
static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
```
Then you can use the `WebClient` in your test. You can also use `testing.runWithSpan(...)` to create spans and `testing.waitAndAssertTraces(...)` to verify that the spans have the correct structure instead of trying to encode trace and span info in the http response. I'd suggest you give it a try, it will be much nicer than what you currently have.

Reply:
ok - I'll abandon this

I don't think it is healthy to have such a high bar for tests.